src/EnergyPlus/BoilerSteam.cc

```diff
@@ -89,7 +89,7 @@ namespace BoilerSteam {
 
     const char *fluidNameSteam = "STEAM";
 
-    PlantComponent *BoilerSpecs::factory(EnergyPlusData &state, std::string const &objectName)
+    BoilerSpecs *BoilerSpecs::factory(EnergyPlusData &state, std::string const &objectName)
```

Comment:
The factory methods should return the concrete type.  The plant structure will store a `PlantComponent*`, but this conversion should happen when the conversion is needed, not as this function exits.  By returning the concrete type, a caller of the factory can use the concrete type without requiring a cast **back** to the concrete type.  This has no functional effect on EnergyPlus itself.